In [1]:
import pandas as pd
import json
import requests

In [2]:
# Set the option to opt-in to the future behavior
pd.set_option('future.no_silent_downcasting', True)

# Foursquare

Send a request to Foursquare with a small radius (1000m) for all the bike stations in your city of choice. 

In [230]:
#retreiving the data from part 1 for vancouver bike stations
bikes_data = pd.read_csv('bikes_data_part1.csv')

In [231]:
bikes_data.head()

bike_station_name  free_bikes   latitude   longitude
0      10th & Cambie          11  49.262487 -123.114397
1    10th & Columbia           3  49.262206 -123.109091
2  10th & Commercial           1  49.261403 -123.070102
3      10th & Fraser           1  49.261510 -123.089413
4   10th & Granville           2  49.262598 -123.138921

Creating lists for storing information that will be recieved from Foursquare API

- ***bike_station_name*** : Name of the bike station
- ***poi_name*** : Name of the point of interest (Bars)
- ***rating*** : Rating of POI
- ***distance_to_station*** : Distance of POI from the bike station
- ***popularity*** : Popularity of POI
- ***total_ratings*** : Total ratings of POI
- ***cc_payment*** : If POI accepts credit card payment
- ***outdoor_seating*** : If POI offers outdoor seating
- ***street_parking*** : If POI offers street parking

In [234]:
bike_station_name =[]
poi_name = []
rating = []
distance_to_station = []
popularity = []
total_ratings = []
cc_payment = []
outdoor_seating = []
street_parking =[]
price_fs = []

In [235]:
# This API calls retrieve the bars around the bike station.

def get_poi(latitude,longitude):
    url = "https://api.foursquare.com/v3/places/search"

    params={
        'radius':1000,
        'll':f'{latitude},{longitude}',
        'categories':'13003,13004,13005,13006,13007,13008,13009,13010,13011,13012,13013,13014,13015,13016,13017,13018,13019,13020,13021,13022,13023,13024,13025,13389'
        }

    headers = {
     "accept": "application/json",
     "Authorization": "fsq3vRYFiHIvZ0kw5BVmIupZkvD8NeBdN8xSNZRDGfu7lcw="
    }

    response = requests.get(url, headers=headers,params=params).json()

    return response


The below API (#2) calls use the Foursquare id of the POI retrieved from the above API (#1) call which results in retrieveing the features of the API

*API CALL #1 : api.foursquare.com/v3/places/search results in businessess search for set of different categories*

*API CALL #2 : api.foursquare.com/v3/places/{i['fsq_id']} results in place search and its features using the foursquare id of the place*

In [ ]:
for index,row in bikes_data.iterrows():
    
    result = get_poi(latitude=row['latitude'],longitude=row['longitude'])
    
    headers = {
        "accept": "application/json",
        "Authorization": "fsq3vRYFiHIvZ0kw5BVmIupZkvD8NeBdN8xSNZRDGfu7lcw="
        }

    for i in result['results']:

        bike_station_name.append(row['bike_station_name'])

        poi_name.append(i['name'])

        distance_to_station.append(i['distance'])

        url = f"https://api.foursquare.com/v3/places/{i['fsq_id']}?fields=stats,rating,features,popularity"
    
        second_response = requests.get(url, headers=headers).json()

        #appending 0 in case of no poularity feature found in response
        if(second_response.get('popularity')):
            popularity.append(second_response['popularity'])
        else:
            popularity.append(0)

        #appending 0 in case of no rating feature found in response
        if(second_response.get('rating')):
            rating.append(second_response['rating'])
        else:
            rating.append(0)

        #appending 0 in case of no total_ratings feature found in response
        if(second_response.get('stats')):
            total_ratings.append(second_response['stats']['total_ratings'])
        else:
            total_ratings.append(0)
        
        #assumption of False in case of no CC feature found in response
        if(second_response.get('features')):
            if(second_response['features'].get('payment')):    
                cc_payment.append(second_response['features']['payment']['credit_cards']['accepts_credit_cards'])
            else:
                cc_payment.append(False)

            #assumption of False in case of no outdoor_seating feature found in response
            #assumption of False in case of no street_parking feature found in response

            if(second_response['features'].get('amenities')):
                if(second_response['features']['amenities'].get('outdoor_seating')):
                    outdoor_seating.append(second_response['features']['amenities']['outdoor_seating'])
                else:
                    outdoor_seating.append(False)

                if(second_response['features']['amenities'].get('street_parking')):
                    street_parking.append(second_response['features']['amenities']['street_parking'])
                else:
                    street_parking.append(False)
                    
            #assumption of False in case of no outdoor_seating feature found in response
            #assumption of False in case of no street_parking feature found in response
            else: 
                outdoor_seating.append(False)
                street_parking.append(False)

        #assumption of False in case of no CC feature found in response
        #assumption of False in case of no outdoor_seating feature found in response
        #assumption of False in case of no street_parking feature found in response
        else:
            outdoor_seating.append(False)
            street_parking.append(False)
            cc_payment.append(False)

In [ ]:
# making sure the length of the lists are same to create a dataframe

print(len(bike_station_name))
print(len(poi_name))
print(len(rating))
print(len(distance_to_station))
print(len(popularity))
print(len(cc_payment))
print(len(outdoor_seating))
print(len(street_parking))
print(len(total_ratings))

Put your parsed results into a DataFrame

In [239]:
fsq_data = pd.DataFrame({'poi_name':poi_name,
              'rating_fs':rating,
              'total_ratings_fs':total_ratings,
              'popularity_fs':popularity,
              'cc_payment_fs':cc_payment,
              'outdoor_seating_fs':outdoor_seating,
              'street_parking_fs':street_parking,
              'bike_station_name':bike_station_name,
              'distance_to_station_fs':distance_to_station
              })

In [240]:
fsq_data

poi_name  rating_fs  total_ratings_fs  popularity_fs  \
0     Rogue Kitchen & Wetbar        8.3               127       0.981731   
1        33 Acres Brewing Co        9.3               380       0.983225   
2           The Cascade Room        8.6               200       0.945325   
3           Biercraft Bistro        8.1               155       0.920526   
4         Faculty Brewing Co        8.1                58       0.950547   
...                      ...        ...               ...            ...   
2222      The Vancouver Club        8.5                42       0.991487   
2223            Tap & Barrel        8.6               353       0.994236   
2224    Pourhouse Restaurant        8.9               381       0.973350   
2225   Steamworks Brewing Co        8.4               919       0.998876   
2226              The Magnet        8.2                15       0.971988   

      cc_payment_fs  outdoor_seating_fs  street_parking_fs  \
0              True                True              False   
1              True               False              False   
2              True               False              False   
3              True                True              False   
4             False               False              False   
...             ...                 ...                ...   
2222          False               False              False   
2223           True                True              False   
2224           True                True              False   
2225           True                True              False   
2226          False               False              False   

                                      bike_station_name  \
0                                         10th & Cambie   
1                                         10th & Cambie   
2                                         10th & Cambie   
3                                         10th & Cambie   
4                                         10th & Cambie   
...                                                 ...   
2222  šxʷƛ̓ənəq Xwtl'e7énḵ Square - Vancouver Art Ga...   
2223  šxʷƛ̓ənəq Xwtl'e7énḵ Square - Vancouver Art Ga...   
2224  šxʷƛ̓ənəq Xwtl'e7énḵ Square - Vancouver Art Ga...   
2225  šxʷƛ̓ənəq Xwtl'e7énḵ Square - Vancouver Art Ga...   
2226  šxʷƛ̓ənəq Xwtl'e7énḵ Square - Vancouver Art Ga...   

      distance_to_station_fs  
0                        250  
1                        674  
2                        984  
3                        744  
4                        994  
...                      ...  
2222                     587  
2223                     765  
2224                     890  
2225                     690  
2226                     689  

[2227 rows x 9 columns]

Replacing the 0 appended values for columns rating_fs, total_ratings_fs, popularity_fs with mean of the columns

In [241]:
fsq_data['rating_fs'] = fsq_data['rating_fs'].replace(0, round(fsq_data['rating_fs'].mean(),1))

fsq_data['total_ratings_fs'] = fsq_data['total_ratings_fs'].replace(0, round(fsq_data['total_ratings_fs'].mean(),0))

fsq_data['popularity_fs'] = fsq_data['popularity_fs'].replace(0, round(fsq_data['popularity_fs'].mean(),0))

In [ ]:
#checking dataframe after the replacement

fsq_data.head()

Checking how many results ( POI ) per bike_station is recieved in foursquare

In [246]:
fsq_data[fsq_data['bike_station_name']=='10th & Cambie']

poi_name  rating_fs  total_ratings_fs  popularity_fs  \
0  Rogue Kitchen & Wetbar        8.3               127       0.981731   
1     33 Acres Brewing Co        9.3               380       0.983225   
2        The Cascade Room        8.6               200       0.945325   
3        Biercraft Bistro        8.1               155       0.920526   
4      Faculty Brewing Co        8.1                58       0.950547   
5             Fox Cabaret        8.0                50       0.962219   
6           Mahony & Sons        7.5                86       0.994642   
7                 Tocador        7.5                14       0.936085   
8     Vancouver Hackspace        7.4                 8       0.855685   
9         The Wicklow Pub        7.1                83       0.988985   

   cc_payment_fs  outdoor_seating_fs  street_parking_fs bike_station_name  \
0           True                True              False     10th & Cambie   
1           True               False              False     10th & Cambie   
2           True               False              False     10th & Cambie   
3           True                True              False     10th & Cambie   
4          False               False              False     10th & Cambie   
5          False               False              False     10th & Cambie   
6          False                True              False     10th & Cambie   
7          False               False              False     10th & Cambie   
8          False               False              False     10th & Cambie   
9           True                True              False     10th & Cambie   

   distance_to_station_fs  
0                     250  
1                     674  
2                     984  
3                     744  
4                     994  
5                     965  
6                     811  
7                     996  
8                     820  
9                     760

In [247]:
# saving the bikes data to csv for Part 3
fsq_data.to_csv('fsq_data.csv', sep=',', index=False, encoding='utf-8')

# Yelp

Send a request to Yelp with a small radius (1000m) for all the bike stations in your city of choice. 

In [253]:
bikes_data.head()

bike_station_name  free_bikes   latitude   longitude
0      10th & Cambie          11  49.262487 -123.114397
1    10th & Columbia           3  49.262206 -123.109091
2  10th & Commercial           1  49.261403 -123.070102
3      10th & Fraser           1  49.261510 -123.089413
4   10th & Granville           2  49.262598 -123.138921

In [275]:
yelp_key = 'Bearer Ohznuk5veifhcVncNy_E0R6tI98TF-DkuUJyICPYMoZaSpiQX_9bzR8kgJGVve9sFNU6aetLykAeB2eHLHyQJ4GP_LGaHIzUyEJ_CKuXYTsw8ucXXFa04NIXTZ0dZnYx'

In [276]:
def get_poi_yelp(latitude,longitude):
    url = f'https://api.yelp.com/v3/businesses/search?radius=1000&latitude={latitude}&longitude={longitude}&categories=bars'

    headers = {
        "accept": "application/json",
        'Authorization':yelp_key
    }

    response = requests.get(url, headers=headers).json()
    print(response)
    return response


Creating lists for storing information that will be recieved from Foursquare API

- ***bike_station_name*** : Name of the bike station
- ***poi_name_y*** : Name of the point of interest (Bars)
- ***rating_y*** : Rating of POI
- ***category*** : Category of the POI
- ***distance_to_station_y*** : Distance of POI from the bike station
- ***price*** : Total price bracket of POI
- ***review_count*** : The review count of POI on yelp

In [277]:
bike_station_name_y = []
poi_name_y = []
rating_y = []
category = []
distance_to_station_y = []
price = []
review_count = []

In [ ]:
for index,row in bikes_data.iterrows():
    
    result = get_poi_yelp(latitude=row['latitude'],longitude=row['longitude'])

    for i in result['businesses']:

        bike_station_name_y.append(row['bike_station_name'])

        poi_name_y.append(i['name'])

        rating_y.append(i['rating'])

        category.append(i['categories'][0]['title'])

        if(i.get('price')):
            price.append(i['price'])
        else:
            price.append(0)

        distance_to_station_y.append(round(i['distance'],2))

        review_count.append(i['review_count'])
        

Put your parsed results into a DataFrame

In [301]:
#checking the lengths of lists are same to create a dataframe

print(len(bike_station_name_y))
print(len(poi_name_y))
print(len(rating_y))
print(len(category))
print(len(distance_to_station_y))
print(len(price))
print(len(review_count))

4148
4148
4148
4148
4148
4148
4148


In [302]:
yelp_data= pd.DataFrame({'poi_name':poi_name_y,
              'rating_yelp':rating_y,
              'category_yelp':category,
              'bike_station_name':bike_station_name_y,
              'distance_to_station':distance_to_station_y,
              'price_yelp':price,
              'review_count_yelp':review_count
              })

In [303]:
yelp_data

poi_name  rating_yelp   category_yelp  \
0        Cactus Club Cafe          3.7  Canadian (New)   
1             Anh and Chi          4.0      Vietnamese   
2        The Cascade Room          3.9            Pubs   
3       Indochine Kitchen          3.2    Asian Fusion   
4       CRAFT Beer Market          3.4        Beer Bar   
...                   ...          ...             ...   
4143            Shizen Ya          4.2        Japanese   
4144  Earls Kitchen + Bar          3.6  Canadian (New)   
4145    Grain Tasting Bar          3.4       Wine Bars   
4146       The Lennox Pub          3.5            Pubs   
4147   Tableau Bar Bistro          4.0            Bars   

                                      bike_station_name  distance_to_station  \
0                                         10th & Cambie               221.48   
1                                         10th & Cambie              1284.21   
2                                         10th & Cambie               992.50   
3                                         10th & Cambie               707.63   
4                                         10th & Cambie              1050.39   
...                                                 ...                  ...   
4143  šxʷƛ̓ənəq Xwtl'e7énḵ Square - Vancouver Art Ga...               395.61   
4144  šxʷƛ̓ənəq Xwtl'e7énḵ Square - Vancouver Art Ga...               296.78   
4145  šxʷƛ̓ənəq Xwtl'e7énḵ Square - Vancouver Art Ga...               207.77   
4146  šxʷƛ̓ənəq Xwtl'e7énḵ Square - Vancouver Art Ga...               247.58   
4147  šxʷƛ̓ənəq Xwtl'e7énḵ Square - Vancouver Art Ga...               523.38   

     price_yelp  review_count_yelp  
0            $$                150  
1           $$$                672  
2            $$                235  
3            $$                114  
4            $$                520  
...         ...                ...  
4143         $$                589  
4144         $$                285  
4145         $$                 54  
4146         $$                137  
4147         $$                287  

[4148 rows x 7 columns]

In [304]:
yelp_data[yelp_data['bike_station_name']=='10th & Cambie']

poi_name  rating_yelp   category_yelp bike_station_name  \
0           Cactus Club Cafe          3.7  Canadian (New)     10th & Cambie   
1                Anh and Chi          4.0      Vietnamese     10th & Cambie   
2           The Cascade Room          3.9            Pubs     10th & Cambie   
3          Indochine Kitchen          3.2    Asian Fusion     10th & Cambie   
4          CRAFT Beer Market          3.4        Beer Bar     10th & Cambie   
5                R&B Brewing          4.1       Breweries     10th & Cambie   
6       Suyo Modern Peruvian          4.6        Peruvian     10th & Cambie   
7    Johnnie Fox's Irish Pub          3.9           Irish     10th & Cambie   
8                R&B Brewing          4.3           Pizza     10th & Cambie   
9             Red Truck Beer          3.9       Breweries     10th & Cambie   
10  Italia Bella Bar & Grill          4.0         Italian     10th & Cambie   
11                   Tocador          3.7  Latin American     10th & Cambie   
12                 Sing Sing          3.7        Beer Bar     10th & Cambie   
13           Mahony’s Tavern          2.7       Irish Pub     10th & Cambie   
14     Lucy's Eastside Diner          3.2          Diners     10th & Cambie   
15  The Wicklow Public House          3.4            Pubs     10th & Cambie   
16                 Key Party          4.4         Lounges     10th & Cambie   
17         Browns Crafthouse          4.0            Bars     10th & Cambie   
18             Good Co. Bars          3.2            Bars     10th & Cambie   
19                  Bar Susu          3.8       Wine Bars     10th & Cambie   

    distance_to_station price_yelp  review_count_yelp  
0                221.48         $$                150  
1               1284.21        $$$                672  
2                992.50         $$                235  
3                707.63         $$                114  
4               1050.39         $$                520  
5                964.10          $                 66  
6               1349.23          0                 29  
7               1395.62         $$                121  
8                943.58         $$                 23  
9               1469.95         $$                 88  
10               602.17         $$                 40  
11               999.09         $$                 50  
12              1014.38         $$                128  
13               783.87         $$                164  
14              1001.02         $$                337  
15               763.76         $$                 90  
16               983.90          0                 27  
17              1186.57          0                 22  
18              1056.93         $$                 83  
19               995.92          0                 26

In [305]:
#assumption of price columns with 0 values atleast offers '$$' - price level of 2 ( medium priced ranged bars)
yelp_data['price_yelp'] = yelp_data['price_yelp'].replace(0,'$$')

In [306]:
## replacing the string variables with integer price levels 

yelp_data['price_yelp'] = yelp_data['price_yelp'].replace({
    '$':1,
    '$$':2,
    '$$$':3,
    '$$$$':4
})

In [307]:
# saving the bikes data to csv for Part 3
yelp_data.to_csv('yelp_data.csv', sep=',', index=False, encoding='utf-8')

# Comparing Results

Which API provided you with more complete data? Provide an explanation. 

*** Comparison between YELP and FOURSQUARE API's***

- There were pros and cons with both the API but Yelp definitely did provide a broader search and large number of results around the bike stations. 

- As compared to Yelp, I had to do a second API call in foursquare to get more details about a 
business place. So 2 foursquare calls = 1 Yelp API call

- The Foursquare API had more interesting features such as payments types accepted, outdoor seating, parking as compared to Yelp.

- I did noticed one thing while overlapping these results that categories ditribution over both the API do differ such as Cactus Club Cafe ( one of the POIs) yielded 164 rows in YELP and none in Foursquare. There might be a reason that it is not listed under 'Bars' in foursquare.

- Both the API radius perimeters resulted in broader search with distance Max values of 1973m and 2626m 

In [308]:
#the data points foursquare retrieved
fsq_data.shape

(2227, 9)

In [309]:
#the data points yelp retrieved
yelp_data.shape

(4148, 7)

In [310]:
#checking the stats for foursquare data
fsq_data.describe()

rating_fs  total_ratings_fs  popularity_fs  distance_to_station_fs
count  2227.000000       2227.000000    2227.000000             2227.000000
mean      7.620117        124.615626       0.925465              596.807364
std       0.853748        147.416489       0.137426              258.763304
min       5.100000          8.000000       0.018031               10.000000
25%       6.700000         29.000000       0.939006              399.000000
50%       7.800000         86.000000       0.966661              613.000000
75%       8.300000        127.000000       0.985684              812.000000
max       9.300000        919.000000       1.000000             1973.000000

In [311]:
#checking the stats for yelp data
yelp_data.describe()

rating_yelp  distance_to_station  review_count_yelp
count  4148.000000          4148.000000        4148.000000
mean      3.884643           648.107387         200.608245
std       0.518422           329.577047         298.560529
min       1.200000             1.970000           1.000000
25%       3.600000           374.825000          34.000000
50%       4.000000           646.345000         114.000000
75%       4.200000           907.077500         285.000000
max       5.000000          2477.070000        2626.000000

In [288]:
fsq_data.isnull().sum()

poi_name                  0
rating_fs                 0
total_ratings_fs          0
popularity_fs             0
cc_payment_fs             0
outdoor_seating_fs        0
street_parking_fs         0
bike_station_name         0
distance_to_station_fs    0
dtype: int64

In [289]:
yelp_data.isnull().sum()

poi_name               0
rating_yelp            0
category_yelp          0
bike_station_name      0
distance_to_station    0
price_yelp             0
delivery_yelp          0
pickup_yelp            0
review_count_yelp      0
dtype: int64

In [290]:
fsq_data['bike_station_name'].nunique()

244

In [291]:
yelp_data['bike_station_name'].nunique()

245

Get the top 10 restaurants according to their rating

### Foursquare Data Top 10 Bars

In [312]:
fsq_data.drop_duplicates(subset=['poi_name']).sort_values(by='rating_fs',ascending=False).head(10)

poi_name  rating_fs  total_ratings_fs  \
1                  33 Acres Brewing Co        9.3               380   
497                         Keefer Bar        9.1               234   
506               Callister Brewing Co        9.0                31   
496               Pourhouse Restaurant        8.9               381   
14                         Anh and Chi        8.9                92   
516  Cardero's Restaurant & Marine Pub        8.7               379   
232                         The Boxcar        8.7                38   
501                         Alibi Room        8.7               441   
2                     The Cascade Room        8.6               200   
520                     Score on Davie        8.6               228   

     popularity_fs  cc_payment_fs  outdoor_seating_fs  street_parking_fs  \
1         0.983225           True               False              False   
497       0.986027           True                True              False   
506       0.995955          False               False              False   
496       0.973350           True                True              False   
14        0.988856          False               False              False   
516       0.978347           True                True              False   
232       0.946197          False               False              False   
501       0.978175           True                True              False   
2         0.945325           True               False              False   
520       0.980303           True                True              False   

    bike_station_name  distance_to_station_fs  
1       10th & Cambie                     674  
497  Abbott & Cordova                     533  
506   Adanac & McLean                     531  
496  Abbott & Cordova                     190  
14    10th & Columbia                     986  
516  Alberni & Jervis                     448  
232    1st & Manitoba                     848  
501  Abbott & Cordova                     510  
2       10th & Cambie                     984  
520  Alberni & Jervis                     787

### Yelp Data Top 10 Bars

In [313]:
yelp_data.drop_duplicates(subset=['poi_name']).sort_values(by='rating_yelp',ascending=False).head(10)

poi_name  rating_yelp  category_yelp  \
3082                Alchemy Bar and Kitchen          5.0        Burgers   
143                      Osteria Elio Volpe          5.0        Italian   
145               Ramson’s Pizzeria And Bar          5.0           Bars   
1114                       Arike Restaurant          4.9        African   
1617                               Bar Gobo          4.8      Wine Bars   
42    Hariyali Express Indian Cuisine & Bar          4.8         Indian   
45                   Giancarlo's Sports Bar          4.8   Coffee & Tea   
290                              The Watson          4.8  Cocktail Bars   
3259           Sports Illustrated Clubhouse          4.7    Sports Bars   
1856                   Bombay Kitchen + Bar          4.7         Indian   

            bike_station_name  distance_to_station price_yelp  \
3082  Olympic Village Station              1054.24          2   
143                10th & Oak              1040.61          2   
145                10th & Oak               680.38          2   
1114           Aquatic Centre               974.52          2   
1617      Campbell & Hastings              1052.27          2   
42          10th & Commercial               571.95          2   
45          10th & Commercial               736.15          1   
290             14th & Fraser               820.07          2   
3259  Pharmaceutical Sciences              1073.94          2   
1856         Chilco & Barclay               360.60          2   

      review_count_yelp  
3082                  1  
143                   1  
145                   1  
1114                 27  
1617                  4  
42                    6  
45                   12  
290                   8  
3259                  3  
1856                  3

In [325]:
yelp_data[yelp_data['poi_name']=='Cactus Club Cafe'	]

poi_name  rating_yelp   category_yelp  \
0     Cactus Club Cafe          3.7  Canadian (New)   
26    Cactus Club Cafe          3.7  Canadian (New)   
69    Cactus Club Cafe          3.7  Canadian (New)   
132   Cactus Club Cafe          3.7  Canadian (New)   
134   Cactus Club Cafe          3.7  Canadian (New)   
...                ...          ...             ...   
4049  Cactus Club Cafe          3.7  Canadian (New)   
4054  Cactus Club Cafe          3.7  Canadian (New)   
4108  Cactus Club Cafe          3.7  Canadian (New)   
4132  Cactus Club Cafe          4.1  Canadian (New)   
4140  Cactus Club Cafe          4.2  Canadian (New)   

                                      bike_station_name  distance_to_station  \
0                                         10th & Cambie               221.48   
26                                      10th & Columbia               597.64   
69                                     10th & Granville               121.66   
132                                          10th & Oak               671.00   
134                                          10th & Oak               976.67   
...                                                 ...                  ...   
4049                                        Wylie & 2nd              1090.79   
4054                        Yaletown-Roundhouse Station               102.31   
4108                                        Yukon & 6th               419.06   
4132  šxʷƛ̓ənəq Xwtl'e7énḵ Square - Vancouver Art Ga...               285.51   
4140  šxʷƛ̓ənəq Xwtl'e7énḵ Square - Vancouver Art Ga...               721.09   

     price_yelp  review_count_yelp  
0             2                150  
26            2                150  
69            2                162  
132           2                150  
134           2                162  
...         ...                ...  
4049          3                218  
4054          3                218  
4108          2                150  
4132          2                492  
4140          2                871  

[164 rows x 7 columns]

In [344]:
fsq_data[fsq_data['poi_name'].str.contains('Cactus')]

Empty DataFrame
Columns: [poi_name, rating_fs, total_ratings_fs, popularity_fs, cc_payment_fs, outdoor_seating_fs, street_parking_fs, bike_station_name, distance_to_station_fs]
Index: []

### Joining YELP and FOURSQUARE

In [350]:
# Merge the DataFrames on the 'poi_name' column
merged_data = pd.merge(yelp_data, fsq_data, on=('poi_name','bike_station_name'), how='inner')

# Display the merged DataFrame
merged_data

poi_name  rating_yelp   category_yelp  \
0    The Cascade Room          3.9            Pubs   
1             Tocador          3.7  Latin American   
2         Anh and Chi          4.0      Vietnamese   
3    The Cascade Room          3.9            Pubs   
4             Tocador          3.7  Latin American   
..                ...          ...             ...   
466      JOEY Burrard          4.1  Canadian (New)   
467         The Ellis          4.2   Cocktail Bars   
468          Corduroy          4.0            Bars   
469  The Cascade Room          3.9            Pubs   
470      JOEY Burrard          4.1  Canadian (New)   

                                     bike_station_name  distance_to_station  \
0                                        10th & Cambie               992.50   
1                                        10th & Cambie               999.09   
2                                      10th & Columbia               995.65   
3                                      10th & Columbia               606.21   
4                                      10th & Columbia               612.81   
..                                                 ...                  ...   
466                        Yaletown-Roundhouse Station               899.52   
467                                         York & Yew                21.68   
468                                         York & Yew               493.23   
469                                        Yukon & 6th               996.98   
470  šxʷƛ̓ənəq Xwtl'e7énḵ Square - Vancouver Art Ga...               205.26   

    price_yelp  review_count_yelp  rating_fs  total_ratings_fs  popularity_fs  \
0            2                235        8.6               200       0.945325   
1            2                 50        7.5                14       0.936085   
2            3                672        8.9                92       0.988856   
3            2                235        8.6               200       0.945325   
4            2                 50        7.5                14       0.936085   
..         ...                ...        ...               ...            ...   
466          2                583        8.2               260       0.992690   
467          2                 84        6.5               106       0.959523   
468          2                 79        8.5                45       0.884569   
469          2                235        8.6               200       0.945325   
470          2                583        8.2               260       0.992690   

     cc_payment_fs  outdoor_seating_fs  street_parking_fs  \
0             True               False              False   
1            False               False              False   
2            False               False              False   
3             True               False              False   
4            False               False              False   
..             ...                 ...                ...   
466           True                True              False   
467          False               False              False   
468           True                True              False   
469           True               False              False   
470           True                True              False   

     distance_to_station_fs  
0                       984  
1                       996  
2                       986  
3                       598  
4                       610  
..                      ...  
466                     930  
467                      10  
468                     493  
469                     987  
470                     219  

[471 rows x 14 columns]

In [346]:
# saving the bikes data to csv for Part 3
merged_data.to_csv('merged_data.csv', sep=',', index=False, encoding='utf-8')